tokio/src/coop.rs

```diff
@@ -45,7 +45,7 @@
 //! other futures. By doing this, you avoid double-counting each iteration of
 //! the outer future against the cooperating budget.
 //!
-//! [`poll`]: https://doc.rust-lang.org/std/future/trait.Future.html#tymethod.poll
+//! [`poll`]: trait@std::future::Future#tymethod.poll
```

Comment:
This seems like it should link directly to the method.

Reply:
See link comparison below.. the method is ok, but just found it replaces
`std/future` with `nightly/core`


The new link:
https://doc.rust-lang.org/nightly/core/future/trait.Future.html#tymethod.poll
the old link
https://doc.rust-lang.org/std/future/trait.Future.html#tymethod.poll

Any thoughts? :)

Reply:
Just checked another similar links, and we have

new
https://doc.rust-lang.org/nightly/std/fs/fn.remove_dir_all.html
old
https://doc.rust-lang.org/std/fs/fn.remove_dir_all.html

this is correct, except ;) it uses nightly, LOL

Reply:
> the method is ok

Actually, you'are right it would be cleaner to make it
`method@std::future::Future::poll`

However, even then we still have other issues, like
1. `std/future` (old) vs `core/future` (new)  
2. `std` (old) vs `nightly` (new)

Reply:
It seems like docs.rs manages to convince rustdoc that they should be links to the stable std. Maybe we can figure out how rustdoc does it?

Reply:
> It seems like docs.rs manages to convince rustdoc that they should be links to the stable

How do you know? :) Just checked an example link from docs.rs which is
`std::fs::File`
from this section
https://docs.rs/tokio/0.2.21/tokio/io/index.html#asyncread-and-asyncwrite

and the link redirects to
https://doc.rust-lang.org/nightly/std/fs/struct.File.html
which is `nightly`.

Reply:
That's just what my memory has told me was the case, but with a closer look, I guess that is wrong. I wonder if there's a flag we can throw at it to have it do so.

Reply:
this looks promising but can't translate it to `cargo doc`

Reply:
anyway, does the `nightly` vs `std` issue block this MR?

Reply:
This works, but the `-Z unstable-options` is a bit uncomfortable.
```
RUSTDOCFLAGS="--cfg docsrs -Z unstable-options --extern-html-root-url std=https://doc.rust-lang.org/stable/" cargo +nightly doc --all-features
```

Reply:
one way or another, switching that should be a separate MR/Issue

Reply:
Regardless of the `std` vs `nightly` issue, the rest two comments I just fixed.

Reply:
No, the `nightly` vs `std` issue doesn't block this PR. It wasn't caused by it, after all.

Reply:
so, should we merge it? :)